packages/grid/x-data-grid/src/components/cell/GridActionsCell.tsx

```diff
@@ -64,10 +64,18 @@ const GridActionsCell = (props: GridActionsCellProps) => {
   }, [hasFocus]);
 
   React.useEffect(() => {
-    if (focusedButtonIndex >= 0) {
-      const child = rootRef.current?.children[focusedButtonIndex] as HTMLElement;
-      child.focus();
+    if (focusedButtonIndex < 0 || !rootRef.current) {
+      return;
+    }
+
+    const numberOfChildren = rootRef.current.children.length;
+    if (focusedButtonIndex >= numberOfChildren) {
+      setFocusedButtonIndex(numberOfChildren - 1);
+      return;
     }
```

Comment:
I don't get why you have this correction of `focusedButtonIndex` compared to `numberOfChildren`, and in another `useEffect` you do a similar check with `numberOfButtons`. 

Maybe one correction with either `numberOfChildren` or `numberOfButtons` is enought

Reply:
I removed `setFocusedButtonIndex` from the code highlighted. There's another effect that updates the state.